src/components/common/Modal.vue

```diff
@@ -0,0 +1,182 @@
+<template>
+  <div class="fixed z-50 top-0 left-0 bg-black bg-opacity-60 h-screen w-screen">
+    <div class="flex justify-center items-center w-full h-full">
+      <form @submit="checkForm" class="flex flex-col justify-between bg-white">
+        <div class="flex justify-between m-5">
+          <span>Edytuj profil</span>
+          <button @click="hideModal">x</button>
+        </div>
+        <div class="flex justify-between items-center mx-12 mb-5">
+          <Avatar :first-name="name" />
+          <div class="flex flex-col justify-between h-24">
+            <Button label="Wrzuć awatar" class="h-10 ml-8" />
+            <Button label="Wrzuć awatar" class="h-10 ml-8" is-warning />
+          </div>
+        </div>
+        <div class="flex flex-col items-center mx-12">
+          <text-input
+            classes="relative w-full mb-10"
+            label="Imię"
+            v-model:value="name"
```

Comment:
FYI, ogarnąłem już dobrze ten v-model przez propsy: v-model="name" jeżeli props nazywa się modelValue, jeżeli inaczej to tak jak jest :)